gapis/api/gles/api/buffer_objects.api

```diff
@@ -267,9 +267,9 @@ cmd void glBufferData(GLenum target, GLsizeiptr size, BufferDataPointer data, GL
   }
   CheckSizeGE!GLsizeiptr(size, 0)
 
-  b.Data = switch (data != null) {
-    case true:  clone(as!u8*(data)[0:size])
-    case false: make!u8(size)
+  b.Data = make!u8(size)
+  if (data != null) {
+    copy(b.Data, as!u8*(data)[0:size])
```

Comment:
What was the reason for this change?

Reply:
`@spy_disabled` and `clone` do not get along well, causing the observation to be dropped. `copy` on the other hand works fine.

Reply:
I'm working on this stuff with the new compiler. Care to elaborate a little? Maybe I can get this fixed.

Reply:
`@spy_disabled` is not really fully implemented. It is partially handled in a few places In the templates. One of the problems is that when evaluating an assignment, if the LHS is `@spy_disabled`, the RHS is not evaluated. This is why `foo = clone(bar...)` drops the observation on `bar` if `foo` is `@spy_disabled` as the clone is not evaluated. Simply evaluating the RHS will lead to other problems, as it would have to be evaluated in a `@spy_disabled` way, for example not issuing an error (maybe) if reading from another `@spy_disabled` source - say from `bar` in the `clone` - it's OK to copy from `@spy_disabled` to `@spy_disabled`, but not to copy from `@spy_disabled` to not-`@spy_disabled` since the data would not be available at trace time.

It seemed overly complex to me to get assignment to behave correctly, and too hacky to add special handling to "assignment with clone RHS", so I did what Vulkan does and dropped the clone shortcut. For tracing it doesn't matter, since the `b = make!u8(size)` gets dropped and the `copy` is only used to create the observation. I didn't think the `clone` is all that much faster during mutate than `make` followed by `copy`.

Reply:
ack